### ISSUE TITLE
CI build broken?

### ISSUE THREAD

**Issue by jrochkind** [ruby-marc/ruby-marc#86]
I think the build is currently failing on master. 

the README has a failed build icon. however clicking on it doesn't take you anywehre useful -- so I think there's a bug in that icon in the README. 

In the [actions history](https://github.com/ruby-marc/ruby-marc/actions), it looks like PR #83 is what is first noted as breaking the build -- both a build of the PR itself pre-merge, and a build of master after the PR was merged. 

But that doesn't necesarily mean #83 broke the build, it may just be the first to register a broken build for another reason. However, #74 merged just in advance does not seem broken.   The jruby jobs that are broken -- github CI can't currently show me a log to tell me why. :(  (Maybe because they are too old? They say "Error: We are currently unable to download the log. Please try again later." but I don't know if they mean that). 

But also, all these PR's... why don't they have the build outcome notated on them like they are supposed to?

Paging @billdueber because you are the one that merged #83, and because you seem to be the closest thing to a responsible maintainer on ruby-marc right now. (I don't currently use MARC in my job, but may in the future, so would liek to keep ruby-marc alive).

**Comment by dmolesUC** [comment]
OK, I installed JRuby and did a little poking around. It looks like under Java 17 (probabably any Java 9+, but that's what I tested) there's a module exports issue in `JRubySTAXReader`, producing failures like this one:

```
TypeError: illegal access on 'next': class org.jruby.javasupport.JavaMethod (in module org.jruby.dist) cannot access class com.sun.org.apache.xerces.internal.impl.XMLStreamReaderImpl (in module java.xml) because module java.xml does not export com.sun.org.apache.xerces.internal.impl to module org.jruby.dist
/Users/david/Work/Scratch/ruby-marc/lib/marc/xml_parsers.rb:414:in `parser_dispatch'
``` 

The same problem is mentioned in [this JRuby issue](https://github.com/jruby/jruby/issues/5509); my guess is it has nothing to do with the code and something to do with an update to the Java and/or JRuby versions used in GitHub CI. I tried the `$JAVA_OPTS` workaround mentioned in that thread and it seemed to fix the problem. I'll see if I can figure out how to get that into a GitHub action, so we can at least get CI working, although I'm not convinced it's a good long-term fix.

**Comment by jrochkind** [comment]
Wow thanks @dmolesUC !

If the problem is caused by the test testing with a newer version of jruby than it used to... then fixing it to test with the older version of JRuby again may be an acceptable solution?

**Comment by dmolesUC** [comment]
IDK, I think it's probably better to support the most recent version? But I'm not a JRuby user, so I can't really say.

On that note, I found another issue that was only a warning on JRuby 9.2 but looks to be an error in CI on JRuby 9.3; see [this commit](https://github.com/dmolesUC/ruby-marc/commit/1d6749e84cba2cf55a679594a9aba1bef88055e1).

I've created a pull request (#87) but I wonder if we should document the JAVA_OPTS workaround in the README? Either that or a JRuby user who knows what they're doing should take a look at JRubySTAXReader and see if there's any possibility of a more permanent fix.
